akka-remote/src/test/scala/akka/remote/artery/OutboundIdleShutdownSpec.scala

```diff
@@ -195,7 +195,7 @@ class OutboundIdleShutdownSpec extends ArteryMultiNodeSpec(s"""
 
         def remoteEcho = system.actorSelection(RootActorPath(remoteAddress) / "user" / "echo")
 
-        val echoRef = remoteEcho.resolveOne(3.seconds).futureValue
+        val echoRef = remoteEcho.resolveOne(remainingOrDefault).futureValue
```

Comment:
Not sure if this will fix it but worth a try.

Reply:
It's still the timeout of a single resolve, just after that remote system was started, shouldn't it rather be an `awaitAssert`-retry logic?

Reply:
As far as I know we should be able to send/receive a message immediately after the ActorSystem has been started. I think many of our tests rely on that.

Reply:
But starting actor system B, and then immediately in actor system A resolve an actor in B - remoting in B might not have completely started up/bound ports etc yet?

Reply:
When the ActorSystem.apply returns it should have initialized and bound remoting. I don't think we have anything async there. From the logs we can also see that the deadLetter doesn't occur immediately, which I would expect if it couldn't connect or if the identify message arrived at the other side but the echo actor wasn't found.

```
2022-02-23T04:10:49.5699744Z [INFO] [02/23/2022 04:10:49.569] [pool-1-thread-1] [ArteryTransport(akka://OutboundIdleShutdownSpec)] Started embedded media driver in directory [/opt/volumes/media-driver/OutboundIdleShutdownSpec-6f396db2-27f9-4dda-a2fa-2e9bf75c76e4]
2022-02-23T04:10:49.5824043Z [INFO] [02/23/2022 04:10:49.581] [pool-1-thread-1] [ArteryTransport(akka://OutboundIdleShutdownSpec)] Remoting started with transport [Artery aeron-udp]; listening on address [akka://OutboundIdleShutdownSpec@localhost:40621] with UID [-1581809473256921188]
2022-02-23T04:10:49.6142936Z [info] OutboundIdleShutdownSpec:
2022-02-23T04:10:49.6148475Z Coroner Thread Count starts at 93 in akka.remote.artery.OutboundIdleShutdownSpec
2022-02-23T04:10:49.6157451Z [info] Outbound streams
2022-02-23T04:10:49.6397800Z [WARN] [02/23/2022 04:10:49.639] [pool-1-thread-1-ScalaTest-running-OutboundIdleShutdownSpec] [akka.actor.ActorSystemImpl(OutboundIdleShutdownSpec-remote-0)] Failed to load JFR flight recorder, falling back to noop. Exception: java.lang.ClassNotFoundException: akka.remote.artery.jfr.JFRRemotingFlightRecorder
2022-02-23T04:10:49.6404184Z [WARN] [02/23/2022 04:10:49.639] [pool-1-thread-1-ScalaTest-running-OutboundIdleShutdownSpec] [akka.remote.RemoteActorRefProvider] Akka Cluster not in use - Using Akka Cluster is recommended if you need remote watch and deploy.
2022-02-23T04:10:49.6671234Z [INFO] [02/23/2022 04:10:49.666] [pool-1-thread-1-ScalaTest-running-OutboundIdleShutdownSpec] [ArteryTransport(akka://OutboundIdleShutdownSpec-remote-0)] Started embedded media driver in directory [/opt/volumes/media-driver/OutboundIdleShutdownSpec-remote-0-6d238e3d-9fbd-413f-afa1-0f9d9acbd985]
2022-02-23T04:10:49.6856710Z [INFO] [02/23/2022 04:10:49.684] [pool-1-thread-1-ScalaTest-running-OutboundIdleShutdownSpec] [ArteryTransport(akka://OutboundIdleShutdownSpec-remote-0)] Remoting started with transport [Artery aeron-udp]; listening on address [akka://OutboundIdleShutdownSpec-remote-0@localhost:40511] with UID [-6342214471546211483]
2022-02-23T04:10:52.9901700Z [INFO] [02/23/2022 04:10:52.987] [pool-1-thread-1-ScalaTest-running-OutboundIdleShutdownSpec] [CoordinatedShutdown(akka://OutboundIdleShutdownSpec-remote-0)] Running CoordinatedShutdown with reason [ActorSystemTerminateReason]
2022-02-23T04:10:53.0266503Z [INFO] [02/23/2022 04:10:52.990] [OutboundIdleShutdownSpec-remote-0-akka.remote.default-remote-dispatcher-12] [akka://OutboundIdleShutdownSpec-remote-0@localhost:40511/system/remoting-terminator] Shutting down remote daemon.
2022-02-23T04:10:53.0269096Z [INFO] [02/23/2022 04:10:52.990] [OutboundIdleShutdownSpec-remote-0-akka.remote.default-remote-dispatcher-12] [akka://OutboundIdleShutdownSpec-remote-0@localhost:40511/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
2022-02-23T04:10:53.0415860Z [WARN] [02/23/2022 04:10:53.041] [OutboundIdleShutdownSpec-akka.actor.default-dispatcher-13] [akka://OutboundIdleShutdownSpec/temp/_user_echo$a] received dead letter: ActorIdentity(None,Some(Actor[akka://OutboundIdleShutdownSpec-remote-0@localhost:40511/user/echo#-393299597]))
2022-02-23T04:10:53.0418883Z [INFO] [akkaDeadLetter][02/23/2022 04:10:53.029] [OutboundIdleShutdownSpec-akka.actor.default-dispatcher-6] [akka://OutboundIdleShutdownSpec/temp/_user_echo$a] Message [akka.actor.ActorIdentity] to Actor[akka://OutboundIdleShutdownSpec/temp/_user_echo$a] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://OutboundIdleShutdownSpec/temp/_user_echo$a] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
2022-02-23T04:10:53.0458824Z [INFO] [02/23/2022 04:10:53.045] [OutboundIdleShutdownSpec-remote-0-akka.actor.internal-dispatcher-3] [ArteryTransport(akka://OutboundIdleShutdownSpec-remote-0)] Stopping embedded media driver in directory [/opt/volumes/media-driver/OutboundIdleShutdownSpec-remote-0-6d238e3d-9fbd-413f-afa1-0f9d9acbd985]
2022-02-23T04:10:53.1704358Z [INFO] [02/23/2022 04:10:53.169] [OutboundIdleShutdownSpec-remote-0-akka.remote.default-remote-dispatcher-7] [akka://OutboundIdleShutdownSpec-remote-0@localhost:40511/system/remoting-terminator] Remoting shut down.
2022-02-23T04:10:53.2618309Z [info] - should be stopped when they are idle *** FAILED *** (3 seconds, 612 milliseconds)
2022-02-23T04:10:53.2620129Z [info]   The future returned an exception of type: akka.actor.ActorNotFound, with message: Actor not found for: ActorSelection[Anchor(akka://OutboundIdleShutdownSpec-remote-0@localhost:40511/), Path(/user/echo)]. (
```

I could be wrong and then we can take the easy path of just retrying since that is not the purpose of this test.